### PR TITLE
🔧 Massive build system cleanup

### LINKED ISSUE #1991: Add 'install' target for `make` (so you can run `make install`)
It would be good to be able to run `make install` so that it copies `RoR` into `<installation prefix>/bin`, etc. This makes it easier for packaging, and is fairly standard.

### PR DESCRIPTION
**Pull request by AnotherFoxGuy**
* Removed `FindXXX.cmake` files (See: https://cgold.readthedocs.io/en/latest/rejected/find-modules.html)
* Made Conan required
* Fixed #1991

### CODE REVIEW

**Comment by ulteq** [comment]
* Removed compile definitions
-> No (we should at least keep `USE_ANGELSCRIPT`, `USE_SOCKETW` and `USE_OPENAL`)
* Made Conan required
-> No (the existing (Conan free) build path is working far to well to completely ditch it)

**Comment by AnotherFoxGuy** [comment]
> @ulteq  No (the existing (Conan free) build path is working far to well to completely ditch it)

https://vector-of-bool.github.io/2018/10/15/pmm.html#fear-of-commitment

**Comment by tritonas00** [comment]
> Made Conan required

no please

**Comment by ulteq** [comment]
* What happens if you leave the project? Nobody of us knows how to maintain Conan / PMM.

* What happens if there are upstream issues with Conan / PMM? Just look at the issues @only-a-ptr already had with Conan so far ... I don't plan on having these.
How many new external tools do we depend on if this gets merged?
bintray? conan.io? pmm? python? ...

* I massively dislike it when tools automatically download and execute code from sources I don't have control of.

* How easy is it to manually recompile the dependencies with slight adoptions if needed?

Don't get me wrong. I do like the concept behind PMM (with Conan).

What's the issue with the existing FindXXX scripts? They don't hurt anyone do they? We can't we keep the old build path for a couple of months, then reevaluate?

And if we really decide to pull the trigger on this, then please after 0.4.8 is out.

**Comment by AnotherFoxGuy** [comment]
> * What happens if you leave the project? Nobody of us knows how to maintain Conan / PMM.

I don't see why you wouldn't be able to learn how Conan recipes work, because Conan has better documentation than our homebrew solution(https://docs.conan.io/en/latest/creating_packages.html)

> * What happens if there are upstream issues with Conan / PMM? Just look at the issues only-a-ptr already had with Conan so far ... I don't plan on having these.

When we switched to the cmake meta-project we also had issues with building RoR.
And the cmake meta-project also had build issues (See #1909 #1908 #1420)

>   How many new external tools do we depend on if this gets merged?
>   bintray? conan.io? pmm? python? ...

PMM for managing conan, and Conan for managing the deps.

> * I massively dislike it when tools automatically download and execute code from sources I don't have control of.

All core deps are maintained by me (https://github.com/AnotherFoxGuy/conan-OGRE https://github.com/AnotherFoxGuy/conan-MyGUI etc...)
Also, the cmake meta-project depends on external cmake build systems.

> * How easy is it to manually recompile the dependencies with slight adoptions if needed?

You still can do that with Conan; just download the recipe, make the changes and run `conan create`

> What's the issue with the existing FindXXX scripts? They don't hurt anyone do they?

The issue with FindXXX scripts is that the grab random packages that may not work with RoR, also defining the correct version is extremely difficult with FindXXX scripts

**Comment by ulteq** [comment]
> #1909 #1908

Both of them were not related with cmake.

> The issue with FindXXX scripts is that they grab random packages that may not work with RoR

That's why we have ror-dependencies, no?

> You still can do that with Conan; just download the recipe, make the changes and run `conan create`

And that will build all dependencies locally?

**Comment by AnotherFoxGuy** [comment]
> That's why we have ror-dependencies, no?

FindXXX scripts prefer system-wide packages over external installed packages (https://github.com/Kitware/CMake/blob/master/Modules/FindOpenAL.cmake#L64-L70)

> And that will build all dependencies locally?

No, only the modified dependency, the others will still be downloaded

**Comment by tritonas00** [comment]
@AnotherFoxGuy is `ninja` a requirement for this?

**Comment by tritonas00** [comment]
i get https://pastebin.com/8ZYQ1GKf at OGRE

**Comment by AnotherFoxGuy** [comment]
@tritonas00 I have never tested RelWithDebInfo, i will check

**Comment by tritonas00** [comment]
Ok i did
```
cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo .
make -j8
make zip_and_copy_resources
```
when i `cd bin` then run `./RoR` it works but when i run outside of that dir (`bin/RoR` for example) it fails (cant find ogre plugins) because `PluginFolder=.` in plugins.cfg

**Comment by tritonas00** [comment]
>The libCaelum issue should be fixed now

Indeed :+1:

**Comment by AnotherFoxGuy** [comment]
@only-a-ptr What do you think about this?

**Comment by AnotherFoxGuy** [comment]
> TL;DR Instead of forcing everyone to use a new package manager, improve the documentation on what is needed to build RoR, and improve the CMake files so they find these requirements.

The issue is that all CMake files are homebrew scripts, with no documentation on how to use/maintain it.

Also, with Conan you won't have issues like this anymore: #1979 #1958 since the deps are built on travis/appveyor and downloaded when you run `conan install`

**Comment by pkillnine** [comment]
That wouldn't work for me because my distro doesn't use the standard filesystem hierarchy (/usr, /bin, etc), it stores each package in it's own directory, and links files directly to other packages' directories (It's GuixSD which uses similar system as NixOS).

The latter issue was fixed in the source code anyways, and the first issue has gone for me for some reason since putting it into a package.
